src/features/contracts/contracts-directory.ts

```diff
@@ -65,12 +65,15 @@ const handleResponse: request.RequestCallback = (err, response) => {
 export const fetchContractsLocally = async (repos: RepositoryInfo[]) => {
 	await Promise.all(
 		repos.map(async (repo) => {
-			// We cast to ReadableStream explicitly because `request.get is of type `request.Request` and it controls whether it is a readable or writable stream internally so it is not typings-compatible with ReadableStream, even though it it functionally equivalent.
-			const get = (request.get(
-				getArchiveLinkForRepo(repo),
-				getRequestOptions(repo),
-				handleResponse,
-			) as unknown) as NodeJS.ReadableStream;
+			// Streaming to a passthrough stream fixes an issue when request.get starts draining before it is passsed to `pipeline`, see https://github.com/request/request/issues/2696, https://github.com/request/request/issues/887
+			const pass = new stream.PassThrough();
+			const get = request
+				.get(
+					getArchiveLinkForRepo(repo),
+					getRequestOptions(repo),
+					handleResponse,
+				)
+				.pipe(pass);
```

Comment:
Can you move this after the `await prepareContractDirectory` and avoid the passthrough buffering the response in memory/keep backpressure support

Reply:
Yeah that would work as well, I just found using the passthrough to be more resilient to timing issues, but any solution would do really.